### PR TITLE
github.com/onsi/ginkgo v1.6.0

### PR DESCRIPTION
**Pull request by liggitt**
Bumps ginkgo to the version preferred by our current level of github.com/onsi/gomega

**What type of PR is this?**
/kind cleanup

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/area code-organization
/sig testing

/cc @dims
for dependency bump

/cc @BenTheElder 
for testing dependency

/cc @cblecker 
for license addition

### CODE REVIEW

**Comment by liggitt** [comment]
/priority important-soon

with this bump, all our dependencies are at their natural versions with the exception of golang/x/{sys,tools} which are at the go1.12 branch levels

**Comment by liggitt** [comment]
/hold
for review by all assigned

**Comment by dims** [comment]
@liggitt looks like we will have 2 fsnotify packages with this change (with same code?)
- gopkg.in/fsnotify.v1
- github.com/fsnotify/fsnotify

Anything we can do to collapse them?

**Comment by liggitt** [comment]
> Anything we can do to collapse them?

We'd have transitive dependencies on both (github.com/spf13/viper/viper.go uses github.com/fsnotify/fsnotify), so not easily

**Comment by dims** [comment]
@liggitt other than that the tomb and tail dependency seem ok (other than a note in https://github.com/kubernetes/node-problem-detector/issues/3 where folks ran into a problem with hpcloud/tail)

**Comment by liggitt** [comment]
> @liggitt other than that the tomb and tail dependency seem ok (other than a note in [kubernetes/node-problem-detector#3](https://github.com/kubernetes/node-problem-detector/issues/3) where folks ran into a problem with hpcloud/tail)

it looks like that's only in use by the ginkgo DebugParallel function, which I don't see us using

File: Godeps/LICENSES [hunk 1]
```diff
@@ -20228,6 +20293,43 @@ SOFTWARE.
 ================================================================================
 
 
+================================================================================
+= vendor/gopkg.in/tomb.v1 licensed under: =
+
+tomb - support for clean goroutine termination in Go.
+
+Copyright (c) 2010-2011 - Gustavo Niemeyer <gustavo@niemeyer.net>
```
Comment (liggitt) [inline Godeps/LICENSES]: hmm... this license doesn't have a recognized name... @cblecker, does the text match a known/acceptable license?

Comment (liggitt) [reply Godeps/LICENSES]: ah, looks to match the bsd 3-clause license, but extra eyes are appreciated

Comment (neolit123) [reply Godeps/LICENSES]: looks like the 3 clause BSD:
https://opensource.org/licenses/BSD-3-Clause

> Redistributions in binary form must reproduce the above copyright notice,
this list of conditions and the following disclaimer in the documentation
and/or other materials provided with the distribution.

is not exactly something that certain a parties want to comply with.
i'd ask the license experts.

Comment (liggitt) [reply Godeps/LICENSES]: we already have 14 other dependencies with this license, fyi

Comment (neolit123) [reply Godeps/LICENSES]: in that case we can assume it's OK.

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

File: Godeps/LICENSES (hunk 1 above)
Comment (BenTheElder) [reply Godeps/LICENSES]: FYI https://github.com/kubernetes/community/blob/b1851114db65091eff064cee2b786826b203c231/contributors/devel/sig-architecture/godep.md#reviewing-and-approving-dependency-changes ->
https://github.com/cncf/foundation/blob/master/whitelist-policy.md#approved-licenses-for-whitelist

3 clause BSD is approved

Comment (neolit123) [reply Godeps/LICENSES]: nice list, thanks.

**Pull request merged**